### PR TITLE
Louisiana Military Pay Exclusion

### LINKED ISSUE #2964: Louisiana Military Pay Exclusion
http://www.legis.la.gov/legis/Law.aspx?d=101760
https://revenue.louisiana.gov/TaxForms/IT540iWEB(2022)D1.pdf

### PR DESCRIPTION
**Pull request by XZ-mabel**
Fixes #2964

### Summary
📱🛠️🌐

Add a new policy parameter `ubi_amount` to the US tax and benefit system. Update the `net_income` function to include the universal basic income in the calculation. Add tests and documentation for the new parameter and function.

> _`is_valid` checks_
> _refactored for clarity_
> _autumn leaves falling_

### Walkthrough
*  Add a new parameter `ubi_amount` to the UBI reform function ()

### CODE REVIEW

File: policyengine_us/parameters/gov/states/la/tax/military_pay_exclusion/max_exclusion.yaml
Comment (PavelMakarchuk) [inline policyengine_us/parameters/gov/states/la/tax/military_pay_exclusion/max_exclusion.yaml:1]: rename to max_amount

File: policyengine_us/parameters/gov/states/la/tax/military_pay_exclusion/max_exclusion.yaml [hunk 1]
```diff
@@ -0,0 +1,16 @@
+description: Louisiana provides provides an exclusion to Louisiana residents who were on active duty in the U.S. armed forces, for up to fifty thousand dollars.
```
Comment (PavelMakarchuk) [inline policyengine_us/parameters/gov/states/la/tax/military_pay_exclusion/max_exclusion.yaml:1]:
```suggestion
description: Louisiana allows for this maximum military pay exclusion amount.
```

File: policyengine_us/parameters/gov/states/la/tax/military_pay_exclusion/max_exclusion.yaml [hunk 2]
```diff
@@ -0,0 +1,16 @@
...
+values:
+  2003-01-01: 30_000
+  2022-01-01: 50_000
+
+metadata:
+ period: year
+ unit: year
```
Comment (PavelMakarchuk) [inline policyengine_us/parameters/gov/states/la/tax/military_pay_exclusion/max_exclusion.yaml:8]:
```suggestion
 unit: currency-USD
```

File: policyengine_us/parameters/gov/states/la/tax/military_pay_exclusion/max_amount.yaml
Comment (PavelMakarchuk) [inline policyengine_us/parameters/gov/states/la/tax/military_pay_exclusion/max_amount.yaml:1]: need changelog

File: policyengine_us/variables/gov/states/la/tax/military_pay_exclusion/la_military_compensation.py
Comment (PavelMakarchuk) [inline policyengine_us/variables/gov/states/la/tax/military_pay_exclusion/la_military_compensation.py:1]: delete

File: policyengine_us/variables/gov/states/la/tax/military_pay_exclusion/la_military_pay_exclusion.py [hunk 3]
```diff
@@ -0,0 +1,21 @@
...
+    unit = USD
+    definition_period = YEAR
+    defined_for = StateCode.LA
+
+    def formula(tax_unit, period, parameters):
+        p = parameters(
+            period
+        ).gov.states.la.tax.military_pay_exclusion
+        compensation = tax_unit(
+            "la_military_compensation", period
```
Comment (PavelMakarchuk) [inline policyengine_us/variables/gov/states/la/tax/military_pay_exclusion/la_military_pay_exclusion.py:17]:
```suggestion
            "military_retirement_pay", period
```

File: policyengine_us/variables/gov/states/la/tax/military_pay_exclusion/la_military_pay_exclusion.py [hunk 4]
```diff
@@ -0,0 +1,21 @@
...
+        )
+        max_amount=p.max_amount
+        amount=compensation > max_amount
+        return where(amount,max_amount, compensation)
```
Comment (PavelMakarchuk) [inline policyengine_us/variables/gov/states/la/tax/military_pay_exclusion/la_military_pay_exclusion.py:21]:
```suggestion
        return min_(max_amount, compensation)
```

File: policyengine_us/variables/gov/states/la/tax/military_pay_exclusion/la_military_pay_exclusion.py [hunk 5]
```diff
@@ -0,0 +1,21 @@
...
+    label = "Louisiana military pay exclusion"
+    unit = USD
+    definition_period = YEAR
+    defined_for = StateCode.LA
+
+    def formula(tax_unit, period, parameters):
+        p = parameters(
+            period
+        ).gov.states.la.tax.military_pay_exclusion
+        compensation = tax_unit(
```
Comment (PavelMakarchuk) [inline policyengine_us/variables/gov/states/la/tax/military_pay_exclusion/la_military_pay_exclusion.py:16]:
```suggestion
        compensation = add(tax_unit, period, ["military_retirement_pay"])
```

File: policyengine_us/variables/gov/states/la/tax/military_pay_exclusion/la_military_pay_exclusion.py [hunk 6]
```diff
@@ -0,0 +1,21 @@
...
+            "la_military_compensation", period
+        )
+        max_amount=p.max_amount
```
Comment (PavelMakarchuk) [inline policyengine_us/variables/gov/states/la/tax/military_pay_exclusion/la_military_pay_exclusion.py:19]:
```suggestion
```

File: policyengine_us/variables/gov/states/la/tax/military_pay_exclusion/la_military_pay_exclusion.py [hunk 7]
```diff
@@ -0,0 +1,21 @@
...
+        amount=compensation > max_amount
```
Comment (PavelMakarchuk) [inline policyengine_us/variables/gov/states/la/tax/military_pay_exclusion/la_military_pay_exclusion.py:20]:
```suggestion
```

File: changelog_entry.yaml
Comment (PavelMakarchuk) [inline changelog_entry.yaml:1]: need changelog

File: changelog.yaml [hunk 8]
```diff
@@ -4390,3 +4390,7 @@
     fixed:
     - Replace several arcane variable names with more descriptive names.
   date: 2023-09-06 18:19:32
+- bump: minor
```
Comment (PavelMakarchuk) [inline changelog.yaml:4393]: delete here - we only add changelog_entry.yaml file - review other closed PR's

File: policyengine_us/parameters/gov/states/la/tax/military_pay_exclusion/max_amount.yaml [hunk 9]
```diff
@@ -0,0 +1,16 @@
+description: Louisiana allows for this maximum military pay exclusion amount.
+values:
+  2003-01-01: 30_000
+  2022-01-01: 50_000
+
+metadata:
+ period: year
+ unit: currency-USD
+ label: Louisiana maximum military pay exclusion amount
```
Comment (PavelMakarchuk) [inline policyengine_us/parameters/gov/states/la/tax/military_pay_exclusion/max_amount.yaml:9]:
```suggestion
 label: Louisiana maximum military pay exclusion max amount 
```

File: policyengine_us/variables/gov/states/la/tax/military_pay_exclusion/la_military_pay_exclusion.py [hunk 10]
```diff
@@ -0,0 +1,16 @@
...
+    entity = TaxUnit
+    label = "Louisiana military pay exclusion"
+    unit = USD
+    definition_period = YEAR
+    defined_for = StateCode.LA
+
+    def formula(tax_unit, period, parameters):
+        p = parameters(period).gov.states.la.tax.military_pay_exclusion
+        compensation = add(tax_unit, period, ["military_retirement_pay"])
+        max_amount = p.max_amount
```
Comment (PavelMakarchuk) [inline policyengine_us/variables/gov/states/la/tax/military_pay_exclusion/la_military_pay_exclusion.py:15]:
```suggestion
```

File: policyengine_us/variables/gov/states/la/tax/military_pay_exclusion/la_military_pay_exclusion.py [hunk 11]
```diff
@@ -0,0 +1,16 @@
...
+        return min_(max_amount, compensation)
```
Comment (PavelMakarchuk) [inline policyengine_us/variables/gov/states/la/tax/military_pay_exclusion/la_military_pay_exclusion.py:16]:
```suggestion
        return min_(p.max_amount, compensation)
```

File: changelog_entry.yaml
Comment (PavelMakarchuk) [inline changelog_entry.yaml:1]: add changelog

File: policyengine_us/variables/gov/states/la/tax/military_pay_exclusion/la_military_pay_exclusion.py [hunk 12]
```diff
@@ -0,0 +1,15 @@
+from policyengine_us.model_api import *
+
+
+class la_military_pay_exclusion(Variable):
+    value_type = float
+    entity = TaxUnit
+    label = "Louisiana military pay exclusion"
+    unit = USD
+    definition_period = YEAR
```
Comment (PavelMakarchuk) [inline policyengine_us/variables/gov/states/la/tax/military_pay_exclusion/la_military_pay_exclusion.py:9]: add reference

File: policyengine_us/parameters/gov/states/la/tax/military_pay_exclusion/max_amount.yaml [hunk 13]
```diff
@@ -0,0 +1,16 @@
+description: Louisiana allows for this maximum military pay exclusion amount.
+values:
+  2003-01-01: 30_000
+  2022-01-01: 50_000
+
+metadata:
+ period: year
+ unit: currency-USD
+ label: Louisiana maximum military pay exclusion max amount
```
Comment (PavelMakarchuk) [inline policyengine_us/parameters/gov/states/la/tax/military_pay_exclusion/max_amount.yaml:9]:
```suggestion
 label: Louisiana military pay exclusion max amount 
```

File: policyengine_us/variables/gov/states/la/tax/military_pay_exclusion/la_military_pay_exclusion.py [hunk 14]
```diff
@@ -0,0 +1,15 @@
...
+    defined_for = StateCode.LA
+
+    def formula(tax_unit, period, parameters):
+        p = parameters(period).gov.states.la.tax.military_pay_exclusion
+        compensation = add(tax_unit, period, ["military_retirement_pay"])
```
Comment (PavelMakarchuk) [inline policyengine_us/variables/gov/states/la/tax/military_pay_exclusion/la_military_pay_exclusion.py:14]:
```suggestion
        military_pay = add(tax_unit, period, ["military_service_income"])
```

File: changelog.yaml [hunk 15]
```diff
@@ -4390,3 +4390,4 @@
     fixed:
     - Replace several arcane variable names with more descriptive names.
   date: 2023-09-06 18:19:32
+
```
Comment (PavelMakarchuk) [inline changelog.yaml:4393]: delete extra line

File: policyengine_us/tests/policy/baseline/gov/states/la/tax/military_pay_exclusion/la_military_pay_exclusion.yaml
Comment (PavelMakarchuk) [inline policyengine_us/tests/policy/baseline/gov/states/la/tax/military_pay_exclusion/la_military_pay_exclusion.yaml:1]: use: military_service_income

File: policyengine_us/variables/gov/states/la/tax/military_pay_exclusion/la_military_pay_exclusion.py [hunk 16]
```diff
@@ -0,0 +1,16 @@
...
+    label = "Louisiana military pay exclusion"
+    unit = USD
+    definition_period = YEAR
+    defined_for = StateCode.LA
+    reference = "http://www.legis.la.gov/legis/Law.aspx?d=101760"
+
+    def formula(tax_unit, period, parameters):
+        p = parameters(period).gov.states.la.tax.military_pay_exclusion
+        military_pay = add(tax_unit, period, ["military_service_income"])
+        return min_(p.max_amount, compensation)
```
Comment (PavelMakarchuk) [inline policyengine_us/variables/gov/states/la/tax/military_pay_exclusion/la_military_pay_exclusion.py:16]:
```suggestion
        return min_(p.max_amount, military_pay)
```

**Review by PavelMakarchuk — APPROVED** [review]

**Review by MaxGhenis — APPROVED** [review]

**Pull request merged**